File: src/WeddingBidders.WebApp/projects/wedding-bidders/src/app/core/models/bidder.ts
export enum BidderType {
  Caterer = 0,
  Photographer = 1,
  MakeUpArtist = 2,
  EventPlanner = 3,
  DiscJockey = 4
}

export interface Bidder {
  bidderId: string;
  firstname: string;
  lastname: string;
  email: string;
  companyName?: string;
  description?: string;
  profileId?: string;
  bidderType: BidderType;
  isApproved: boolean;
}

export interface BidderTypeDto {
  name: string;
  value: number;
}

export interface RegisterBidderRequest {
  firstname: string;
  lastname: string;
  email: string;
  password: string;
  confirmPassword: string;
  companyName?: string;
  description?: string;
  bidderType: BidderType;
}

export interface RegisterBidderResponse {
  success: boolean;
  message: string;
  bidderId?: string;
}

File: services/bidding/WeddingBidders.Bidding.Core/Model/Bidder.cs
using WeddingBidders.Shared.Core;

namespace WeddingBidders.Bidding.Core.Model;

public class Bidder : BaseEntity
{
    public Guid BidderId { get; set; }
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string? CompanyName { get; set; }
    public string? Description { get; set; }
    public Guid ProfileId { get; set; }
    public BidderType BidderType { get; set; }
    public bool IsApproved { get; set; }
    public ICollection<Gallery> Galleries { get; set; } = new List<Gallery>();
    public ICollection<Bid> Bids { get; set; } = new List<Bid>();
}

public enum BidderType
{
    Photographer = 0,
    Videographer = 1,
    Caterer = 2,
    Florist = 3,
    DJ = 4,
    Band = 5,
    Planner = 6,
    Decorator = 7,
    Other = 8
}
